### PR TITLE
Fix order search in different networks

### LINKED ISSUE #130: Search for an order in different network does not work
1. Copy an orderID from Rinkeby
2. Switch to Gnosis chain
3. Navigate to the Home page
4. Search for the copied order 

**AR**: failed to fetch error appears, then user is navigated to the not found page

Video: https://watch.screencastify.com/v/ceUQyshRjauwJIuodbMG

**ER**: order should be found, user should be switched to the required network 

**Note**: search for TxHash works fine

### LINKED ISSUE #138: No TxHash in an order when open last batch link
Related to #137

1. Open Home page
2. Press on the Last batch ID 
3. Open order details

**AR**: no TX hash for this order

**ER**: TxHash should present

### PR DESCRIPTION
**Pull request by ramirotw**
# Summary

Close #130 
Close #138

# To Test

1. Copy an orderID from Rinkeby
2. Switch to Gnosis chain
3. Navigate to the Home page
4. Search for the copied order

### CODE REVIEW

File: src/api/operator/operatorApi.ts [hunk 1]
```diff
@@ -71,12 +71,14 @@ export async function getOrder(params: GetOrderParams): Promise<RawOrder | null>
...
   if (!cowInstance || !cowInstanceDev) return null
 
+  let order = null
   try {
-    const order = await cowInstance.cowApi.getOrder(orderId)
-    return order
+    order = await cowInstance.cowApi.getOrder(orderId)
   } catch (error) {
-    return cowInstanceDev.cowApi.getOrder(orderId)
+    order = await cowInstanceDev.cowApi.getOrder(orderId)
```
Comment (matextrem) [inline src/api/operator/operatorApi.ts:78]: Could we move it to finally statement since we are using it both in try and catch ones.

**Review by matextrem — CHANGES_REQUESTED** [review]

**Comment by ramirotw** [comment]
@elena-zh re-posting here your feedback from https://github.com/cowprotocol/explorer/issues/130#issuecomment-1162996669

> Hey @ramirotw , great changes!
However, I have noticed, that order TxHash does not appear when open a transaction for the 1st time using search in a different network. See the video: https://watch.screencastify.com/v/nuo4Q7xONKJ608sQ1UTj (the issue itself starts from 25s on the video)
Could you please take a look into this issue?

**Comment by elena-zh** [comment]
seems that the fix was reverted
image

**Review by matextrem — APPROVED** [review]
LGTM 👍

**Comment by henrypalacios** [comment]
It does not work for me!

When I search on GC the tx `0x5dd8cab9849abc9ff04230d3c7432fa9206fd5b47fd5cb704bc04775f4ec65a2b638eda8c10005cb456b1913ad60a8f3d6a927146144b0ea`  I got :point_down: 
Selection_583


It's a valid order on [mainnet](https://pr119--explorer.review.gnosisdev.com/orders/0x5dd8cab9849abc9ff04230d3c7432fa9206fd5b47fd5cb704bc04775f4ec65a2b638eda8c10005cb456b1913ad60a8f3d6a927146144b0ea)

**Review by alfetopito — APPROVED** [review]

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @ramirotw , I can't load  a TX details page when I search for Rinkeby transaction in Ethereum. See the video: https://watch.screencastify.com/v/DRq2z7sKeGORMI8qYDeT

cant load

Refreshing the page helps, though.

Could you please take a look at it? 

Here were my steps: 
1. Open user details in Rinkeby
2. Copy an order ID from the list
3. Open Home page in Ethereum
4. Search for a copied order ID

**Comment by ramirotw** [comment]
> Hey @ramirotw , I can't load a TX details page when I search for Rinkeby transaction in Ethereum. See the video: https://watch.screencastify.com/v/DRq2z7sKeGORMI8qYDeT

This was caused by the hook that fetches the tokens. The tokens fetch was triggered while still on the mainnet and set a loading flag to true, but when on rinkeby it didn't re-run, so it stayed on loading indefinitely.

**Comment by elena-zh** [comment]
Hey @ramirotw !
The issue is partially fixed, as search for: 
 - Rinkeby transaction in GC- works fine
 - Mainnet transaction in GC 0 works fine
 - GC transaction in Rinkeby - works fine.
BUT search for
 - GC transaction in Mainnet - not working
 - Rinkeby transaction  in Mainnet- not working
 - Mainnet transaction in Rinkeby - not working

Besides, I have found a strange issue with orders list pagination in this Pr: https://watch.screencastify.com/v/YSoUCU7TRfQcyJ0yElph

Thanks!

**Comment by elena-zh** [comment]
@ramirotw , search for orders IDs look good to me now!

However, amounts for TX details are not loaded in this PR when search for transactions from different networks. Could you please take a look at it, as the issue is not reproducible in #137

**Comment by elena-zh** [comment]
> @ramirotw , search for orders IDs look good to me now!
> 
> However, amounts for TX details are not loaded in this PR when search for transactions from different networks. Could you please take a look at it, as the issue is not reproducible in #137 image

Steps to reproduce it: 
1. Copy a Tx hash of a filled order in Gnosis chain
2. Open Home page in Ethereum
3. Paste the copied TxHash from GC into the search field and search for it. 

**AR**: Tx details page is opened, but amounts are not loaded

**Comment by ramirotw** [comment]
@elena-zh please re-test

**Comment by ramirotw** [comment]
> @ramirotw , order search does not work at all (order from a different network)

A commit didn't apply while rebasing 🤦‍♂️, my bad sorry

**Review by elena-zh — APPROVED** [review]

**Pull request merged**